### ISSUE TITLE
Allow us to use With Me on constructor with readonly variables

### ISSUE THREAD

**Issue by JustNrik** [dotnet/vblang#338]
Sometimes constructors got many variables and we have to type Me. all the time
```vb
Sub New(Value() As Object)
    Me.Prop1 = Value(0)
    Me.Prop2 = Value(1)
    Me.Prop3 = Value(2)
    .
    .
    .
    Me.PropN = Value(N-1)
End Sub
```
I would like if we could use With Me instead, for some of us it would look better
```vb
Sub New(Value() As Object)
    With Me
        .Prop1 = Value(0)
        .Prop2 = Value(1)
        .Prop3 = Value(2)
        .
        .
        .
        .PropN = Value(N-1)
    End With
End Sub
```
Currently, we can't compile as it will say that ReadOnly variables cannot be target of an assignment

**Comment by JustNrik** [comment]
@KathleenDollard Have you take a look on this? 👀

**Comment by KathleenDollard** [comment]
This seems reasonable to me, although I haven't explored why Me isn't working here. I assume it is a general restriction on the use of Me in With? (not just in constructors, although that is where it is painful).

**Comment by JustNrik** [comment]
> 
> 
> This seems reasonable to me, although I haven't explored why Me isn't working here. I assume it is a general restriction on the use of Me in With? (not just in constructors, although that is where it is painful).

It only happens with `ReadOnly` fields/properties on the ctor. Everywhere else `With Me` works fine, even in the ctor with non-ReadOnly fields/properties.

**Comment by KathleenDollard** [comment]
Got it. Earlier response was pre-coffee. 

Still seems reasonable.

**Comment by pricerc** [comment]
I agree that this seems reasonable, and it seems a slightly arbitrary limitation.

However, the OP's example isn't the most compelling, since in that example, the 'Me.' is optional anyway. You only need the 'Me.' if the property name is the same as the parameter name in the constructor: e.g. 

``` VB
Class Foo
    Public ReadOnly Property Bar as string
	
    Sub New(bar As String)
        Me.Bar = bar
    End Sub
End Class
```

And of course, if you don't have setters doing something special, there's always the option of directly accessing the backing store, which also removes the need for 'Me': 
``` VB
Class Foo
    Public ReadOnly Property Bar As String

    Sub New(bar As String)
        _Bar = bar
    End Sub
End Class
```
